components/script/dom/bindings/typedarrays.rs

```diff
@@ -142,3 +161,35 @@ where
         TypedArray::from(dest.get())
     }
 }
+
+pub fn create_new_external_array_buffer<T>(
+    cx: JSContext,
+    mapping: Arc<Mutex<Vec<T::Element>>>,
+    offset: usize,
+    range_size: usize,
+    m_end: usize,
+) -> HeapTypedArray<T>
+where
+    T: TypedArrayElement + TypedArrayElementCreator,
+    T::Element: Clone + Copy,
+{
+    /// `freeFunc()` must be threadsafe, should be safely callable from any thread
+    /// without causing conflicts or unexpected behavior.
+    /// <https://github.com/servo/mozjs/blob/main/mozjs-sys/mozjs/js/public/ArrayBuffer.h#L89>
+    unsafe extern "C" fn free_func(_contents: *mut c_void, free_user_data: *mut c_void) {
+        let _ = Arc::from_raw(free_user_data as _);
+    }
+
+    unsafe {
+        let array_buffer = NewExternalArrayBuffer(
+            *cx,
+            range_size as usize,
+            mapping.clone().lock().unwrap().borrow_mut()[offset as usize..m_end as usize]
```

Comment:
I've checked locally, and it compiles if you cut the borrow earlier, as in:

```
let ptr = mapping.lock().unwrap().borrow_mut()[offset as usize..m_end as usize]
                .as_mut_ptr();
        let array_buffer = NewExternalArrayBuffer(
            *cx,
            range_size as usize,
            ptr as _,
            Some(free_func),
            Arc::into_raw(mapping) as _,
        );
```

Reply:
Indeed thanks, addressed